test/core.jl

```diff
@@ -77,6 +77,54 @@ H = similar(A, Float64, 1,1,1)
 # Size
 @test size(A, 1) == size(A, Axis{1}) == size(A, Axis{:row}) == size(A, Axis{:row}())
 
+# Convert between arrays with similar axes
+# first the no-op kind
+A = AxisArray(reshape(1:24, 2, 3, 4), :a, :b, :c)
+@test @inferred(convert(typeof(A), A)) === A
+
+# then change the underlying array type, retaining the axes
+Adense = AxisArray(zeros(Int16, size(A)), axes(A))
+@test typeof(@inferred(convert(typeof(Adense), A))) == typeof(Adense)
+@test axes(convert(typeof(Adense), A)) == axes(A)
+@test all(convert(typeof(Adense), A).data .== A.data)
```

Comment:
I know these are just tests, but this could be written simply and more efficiently as `@test convert(typeof(Adense), A).data == A.data`